app/src/main/java/com/amaze/filemanager/filesystem/root/ListFilesCommand.kt

```diff
@@ -124,6 +124,12 @@ object ListFilesCommand : IRootCommand() {
                         PreferencesConstants.PREFERENCE_ROOT_LEGACY_LISTING,
                         false
                     )
+            // #3476: Check current working dir, change back to / before proceeding
+            runShellCommand("pwd").run {
```

Comment:
How did you test this? There was a reason `/` was replaced in line 115. A better approach will be to just not replace `/` in 115 but needs testing with different su binaries.

Reply:
The code combination so far it works on both Magisk's `busybox` by 0smosis and `toybox`.

> Also, here you're always cd to root / regardless of user navigate to any directory. Won't this restrict user to navigation to any directory other than the root?

Amaze always run `ls` or `stat` with full paths, so there will not be problem of always start from `/`.

Reply:
I don't like this approach. Basically every time we list we're going to run two commands, doesn't seem right, even if it does the job, this is adding a overhead.
What happens if we just remove line 115? Is there any issue?

Reply:
OK, fair enough. That was a trick to prevent sending a `//` to root commands. Should have used an easier way.

And on the overhead of `pwd` and `cd /`, unless we are sure the root shell is used throughout the whole lifecycle of the app, we can't be sure if the second time root shell brings up its `pwd` is staying the last session's ones.